service/dps/storm/topologies/ic/ic-topology/src/main/java/eu/europeana/cloud/service/dps/storm/topologies/ic/topology/api/ImageConverterServiceImpl.java

```diff
@@ -53,58 +59,42 @@
      */
     @Override
     public void convertFile(StormTaskTuple stormTaskTuple) throws IOException, MimeTypeException, MCSException, ICSException, RuntimeException {
-        converterContext = new ConverterContext(new KakaduConverterTiffToJP2());
-        LOGGER.info("The converting process for file " + stormTaskTuple.getFileUrl() + " started successfully");
+        String fileUrl = stormTaskTuple.getFileUrl();
+        LOGGER.info("The converting process for file " + fileUrl + " started successfully");
         String folderPath = null;
 
         try {
             ByteArrayInputStream inputStream = stormTaskTuple.getFileByteDataAsStream();
             if (inputStream != null) {
-                Map<String, String> urlParams = FileServiceClient.parseFileUri(stormTaskTuple.getFileUrl());
-                String fileName = urlParams.get(ParamConstants.P_FILENAME);
-                String cleanName = FilenameUtils.removeExtension(fileName);
-                String inputExtension = ExtensionHelper.getExtension(TaskTupleUtility.getParameterFromTuple(stormTaskTuple, PluginParameterKeys.MIME_TYPE));
-                long randomValue = UUID.randomUUID().getMostSignificantBits();
-                String tempFileName = String.valueOf(randomValue);
-                folderPath = persistStreamToTemporaryStorage(inputStream, tempFileName, inputExtension);
-                String inputFilePath = buildFilePath(folderPath, tempFileName, inputExtension);
+                String inputMimeType = TaskTupleUtility.getParameterFromTuple(stormTaskTuple, PluginParameterKeys.MIME_TYPE);
+                assert (TIFF_MIME_TYPE.equals(inputMimeType) || JPEG_MIME_TYPE.equals(inputMimeType));

```

Comment:
Checking of assertions have to be switched on in JVM settings. Better would be to use if statement. 

http://stackoverflow.com/questions/3018683/what-does-the-assert-keyword-do